### PR TITLE
En passant

### PR DESCRIPTION
**Pull request by MelsenC**
I wrote two methods - first_move? and en_passant?, which I believe check for en_passant adequately. Still need to run through this with Dennis and write tests.

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -58,6 +60,12 @@ def obstructed?(new_x, new_y)
 
   private
 
+  def increment_move
```
Comment (MelsenC) [inline app/models/piece.rb]: Rubocop doesn't like me using the increment! method because it says it skips validations. What's a better way to write this?

Comment (mscoutermarsh) [inline app/models/piece.rb]: Not sure - usually Rubocop docs will make a suggestion, whats the error say?

Comment (MelsenC) [inline app/models/piece.rb]: `app/models/piece.rb:66:5: C: Avoid using increment! because it skips validations.`

Comment (MelsenC) [inline app/models/piece.rb]: I didn't find anything in the docs, searching for increment.

Comment (mscoutermarsh) [inline app/models/piece.rb]: See http://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Rails/SkipsModelValidations

Try doing something like, `game.update_attributes(move_number: move_number+1)`

Comment (MelsenC) [inline app/models/piece.rb]: thanks! 😃

**Pull request closed without merging**